### DISCUSSION TITLE
Create bugs and features in GitHub

### DISCUSSION THREAD

**Discussion by wolfgang-ch** [mytourbook/mytourbook#492] [General]
Until now, bugs and features could be created and tracked with

https://sourceforge.net/p/mytourbook/bugs/
https://sourceforge.net/p/mytourbook/feature-requests/

I would like, that new bugs and features are created only in https://github.com/mytourbook/mytourbook/issues, what do you think?

I don't know if existing bugs and features could be easily transferred to github, so existing bug's and features will still be tracked as before.

**Comment by ziegenberg** [comment]
Sounds like a good plan to have everything in one place!

**Comment by FJBDev** [comment]
>what do you think?

I think it's a good idea to have everything in 1 place.
But I wonder if users will like to have to register to one more site (especially a site targeted for developers mainly) in order to create issues/bugs....
1 site to open a thread (Sourceforge), 1 site to create issues (Github)....

>I don't know if existing bugs and features could be easily transferred to github

There is [this tool](https://github.com/cmungall/gosf2github) available

**Comment by telemaxx** [comment]
1 place is best.

i think discussions could also be done on github.

**Reply by wolfgang-ch** [reply]
Discussions needs an adjusted perl script because discussion and bug/feature requests have a different structure.

I'm working on it and I'm learning now also a bit of perl

-> I just learned how to sort strings instead of numbers :-)

```
C:\sf-2-gh>perl sf-2-gh-script-discussion.pl --assignee wolfgang-ch --usermap sf-2-gh-users.json --collaborators collaborators---mt-from-sf-to-gh.json --token ??? --repo mt-from-sf-to-gh/discussion2 --sf-tracker mytourbook/discussion/622811 --label Discussion-from-SourceForge --dry-run sf-discussion.json
TICKET JSON: sf-discussion.json
Time: 2006-10-13 15:05:08 - Subject: Welcome to Open Discussion
Time: 2007-08-12 15:18:26 - Subject: Mac Support
Time: 2007-10-15 21:29:19 - Subject: What type of data can you import?
Time: 2007-11-07 10:23:12 - Subject: New Hac4 Pro data format
Time: 2007-11-10 21:35:53 - Subject: Mytourbook and Leopard
Time: 2007-12-07 11:05:50 - Subject: Power calculation
Time: 2008-01-07 09:55:33 - Subject: Checkout Source Code
Time: 2008-01-16 14:14:15 - Subject: Tour File example
Time: 2008-03-04 16:34:19 - Subject: Tour Catalogue
Time: 2008-03-07 07:54:38 - Subject: Garmin developer Hompage
Time: 2008-03-22 08:25:24 - Subject: Class definition not found on Leopard
Time: 2008-04-21 08:25:01 - Subject: Change of altitude
Time: 2008-04-27 20:09:56 - Subject: Bitte um Kontakt

...

Time: 2021-12-11 09:10:16.548000 - Subject: Update site
Time: 2021-12-12 15:02:36.224000 - Subject: Different average pace/speed on tour book vs tour info
Time: 2021-12-14 22:20:06.829000 - Subject: Unable to start (SQLException)
Time: 2022-01-01 17:43:29.551000 - Subject: Latest update to v21.12.0 Win64 MSI has destroyed my profile and tour data
Time: 2022-01-15 18:11:56.929000 - Subject: Allow Power to be shown as a regular field when showing values like speed and cadence
Time: 2022-01-16 12:21:05.893000 - Subject: Difficulties with synchronizing to SuuntoApp
Time: 2022-01-16 16:34:48.036000 - Subject: Adjust altitude only for certain parts of track
Time: 2022-01-19 16:19:05.166000 - Subject: Modifier les valeurs de tempΘrature
Time: 2022-01-26 20:19:41.971000 - Subject: Passage de la version 20.3.0 α la mytourbook-with-java-runtime-22.1.0-win-setup
Time: 2022-03-11 20:07:43.828000 - Subject: deleted tours appear with next import
Time: 2022-03-16 23:02:47.987000 - Subject: tourId creation rationale
Time: 2022-03-18 18:49:40.708000 - Subject: Synchronize map with vertical sliders
Time: 2022-03-30 15:51:12.257000 - Subject: Feature Requests - Create Ticket
Time: 2022-04-08 16:34:46.893000 - Subject: Issue with Sporttracks data import into MTB
Time: 2022-04-09 20:56:28.981000 - Subject: MTB 22.3.0 missing icon for "Weather-Showers-Scattered"
Time: 2022-04-10 15:26:34.813000 - Subject: Update Mytourbook auf Version 22.3.0 Fehlermeldung im Tagging-Bereich
Time: 2022-04-11 14:58:30.047000 - Subject: fehlerhafte Suchfunktion und Import Wegpunkte / incorrect search function and import waypoints
Time: 2022-04-12 07:10:13.069000 - Subject: Weather Retrieval
Time: 2022-04-12 09:00:22.994000 - Subject: 3D Map: Darstellungsfehler bei hoher Aufl÷sung
Time: 2022-04-14 09:52:23.603000 - Subject: Impossible importer d'une montre Garmin 305
Time: 2022-04-14 10:36:33.557000 - Subject: Verbesserungsvorschlag Sortierungsm÷glichkeit ⁿber die ▄berschrift im Tour Editor
Died at sf-2-gh-script-discussion.pl line 79, <> line 57112.
```

**Comment by mff73** [comment]
Hello, I am Mick from SourceForge MTB forum.
As I also have a github account (even if not a developper) I will try to create a bug here and comment on the common user of MTB point of view. Let see

**Comment by mff73** [comment]
22.3 installed, found a "bug", and open a ticket.
Just can't find how to add a label :(

**Reply by wolfgang-ch** [reply]
> 22.3 installed, found a "bug", and open a ticket. Just can't find how to add a label :(

There are almost endless access settings for GitHub repos https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization 

-> A "normal" user do not have "Apply/dismiss labels"

**Comment by wolfgang-ch** [comment]
The migrated bugs and features can be tested in these repos

https://github.com/mt-from-sf-to-gh/bugs/issues
https://github.com/mt-from-sf-to-gh/features/issues

When these issues are merged into the MT repo, then they are separated only by tags.

The migration is not perfect but each issue contains a link to the original bug/feature

**Comment by wolfgang-ch** [comment]
Features and bugs are now merged from Sourceforce to GitHub, they can be accessed with these labels

https://github.com/mytourbook/mytourbook/labels/Feature%20from%20SourceForge https://github.com/mytourbook/mytourbook/labels/Bug%20from%20SourceForge

**Comment by wolfgang-ch** [comment]
Help is needed to migrate discussion from SF to GH:

https://sourceforge.net/p/mytourbook/discussion/622811/thread/e95fac98e8
